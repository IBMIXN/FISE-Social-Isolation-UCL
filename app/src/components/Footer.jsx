import React from "react";
import { Flex, Stack, Icon, Text } from "@chakra-ui/core";

const Footer = (props) => (
  <Flex
    as="footer"
    py="4rem"
    direction="column"
    justify="center"
    align="center"
    w="100vw"
    overflowX="hidden"
  >
    <Text textAlign="center">FISE Lounge was created by students at UCL</Text>
    <Stack>
      <a href="https://ibm.com" aria-label="ibm">
        <Icon name="ibm" color="grey.900" w="200px" h="50px" m="1rem" />
      </a>
      <a href="https://nhs.uk" aria-label="nhs">
      <Icon name="nhsX" color="grey.900" w="200px" h="50px" m="1rem" />
      </a>
      <a href="https://ucl.ac.uk" aria-label="ucl">
        <Icon name="ucl" color="gray.700" w="200px" h="50px" m="1rem" />
      </a>
    </Stack>
    <Stack p="1rem" color="gray.400">
      <Text textAlign="center">
        In association with IBM, NHS & University College London.
      </Text>
      <Text textAlign="center">
        Supervised by John McNamara, Dean Mohamedally.
      </Text>
    </Stack>
  </Flex>
);

export default Footer;
