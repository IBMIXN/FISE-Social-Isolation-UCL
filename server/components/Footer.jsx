import { Flex, Stack, Icon, Text, Image } from "@chakra-ui/core";

export const Footer = (props) => (
  <Flex
    as="footer"
    py="8rem"
    direction="column"
    justify="center"
    align="center"
    bg="gray.800"
    w="100vw"
    overflowX="hidden"
    {...props}
  >
    <Text textAlign="center">FISE 2021 by UCL and IBM</Text>
    <Stack>
      <a href="https://ibm.com" aria-label="ibm">
        <Icon name="ibm" color="grey.300" w="200px" h="50px" m="1rem" />
      </a>
      <a href="https://nhs.uk" aria-label="nhs">
        <Icon name="nhsX" color="gray.400" w="200px" h="60px" m="1rem" />
      </a>
      <a href="https://ucl.ac.uk" aria-label="ucl">
        <Icon name="ucl" color="gray.400" w="200px" h="50px" m="1rem" />
      </a>
    </Stack>
    <Stack p="1rem" color="gray.300">
      <Text textAlign="center">
        In association with IBM, NHS & University College London
      </Text>
      <Text textAlign="center">
        Supervised by John McNamara, Dean Mohamedally.
      </Text>
      <Text textAlign="center" color="gray.500">
        DISCLAIMER
      </Text>
      <Text
        textAlign="center"
        maxWidth="35em"
        color="gray.500"
        wordBreak="true"
      >
        The software is an early proof of concept for development purposes and
        should not be used as-is in a live environment without further
        redevelopment and/or testing. No warranty is given and no real data or
        personally identifiable data should be stored. Usage and its liabilities
        are your own.
      </Text>
    </Stack>
  </Flex>
);
