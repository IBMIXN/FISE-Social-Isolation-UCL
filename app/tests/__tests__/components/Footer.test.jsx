import React from 'react';
import ReactDOM from 'react-dom';
import Footer from '../../../src/components/Footer';

import Enzyme, { shallow, render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() })

it('footer render component unit test', () => {
  const wrapper = shallow(<Footer/>)
  
  expect(toJson(wrapper)).toMatchSnapshot();
});

// it('integration test', () => {
//   const wrapper = mount(<Footer />)

//   expect(toJson(wrapper)).toMatchSnapshot();
// });