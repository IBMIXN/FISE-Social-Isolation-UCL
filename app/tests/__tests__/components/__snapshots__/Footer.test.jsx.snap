// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`footer render component unit test 1`] = `
<Flex
  align="center"
  as="footer"
  direction="column"
  justify="center"
  overflowX="hidden"
  py="4rem"
  w="100vw"
>
  <Text
    textAlign="center"
  >
    FISE Lounge was created by students at UCL
  </Text>
  <ForwardRef>
    <a
      href="https://ibm.com"
    >
      <Icon
        color="grey.900"
        h="50px"
        m="1rem"
        name="ibm"
        w="200px"
      />
    </a>
    <a
      href="https://nhs.uk"
    >
      <Image
        h="81px"
        mx="auto"
        src="/icons/nhs-inx-logo.png"
        w="157px"
      />
    </a>
    <a
      href="https://ucl.ac.uk"
    >
      <Icon
        color="gray.700"
        h="50px"
        m="1rem"
        name="ucl"
        w="200px"
      />
    </a>
  </ForwardRef>
  <ForwardRef
    color="gray.400"
    p="1rem"
  >
    <Text
      textAlign="center"
    >
      In association with IBM, NHS & University College London.
    </Text>
    <Text
      textAlign="center"
    >
      Supervised by John McNamara, Dean Mohamedally.
    </Text>
  </ForwardRef>
</Flex>
`;
